File: DrawWithFire/DrawWithFire/ViewController.m
//
//  ViewController.m
//  DrawWithFire
//
//  Created by Quang Tran on 5/13/17.
//  Copyright © 2017 Quang Tran. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()


@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
}

- (void)touchesMoved:(NSSet *)touches withEvent:(UIEvent *)event {
  [fireView setEmitterPositionFromTouch: [touches anyObject]];
}

- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
  [fireView setEmitterPositionFromTouch: [touches anyObject]];
  [fireView setIsEmitting:YES];
}

- (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event {
  [fireView setIsEmitting:NO];
}

- (void)touchesCancelled:(NSSet *)touches withEvent:(UIEvent *)event {
  [fireView setIsEmitting:NO];
}


@end

File: CustomAlert/CustomAlert/ViewController.swift
//
//  ViewController.swift
//  CustomAlert
//
//  Created by Quang Tran on 11/28/16.
//  Copyright © 2016 Quang Tran. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
  }

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }

  @IBAction func showButtonDidTap(sender: AnyObject) {
    let confirmationAlertVC = ConfirmationAlertVC()
    confirmationAlertVC.alertTitle = "Does your video contain only 1 made dunk?"
    
    confirmationAlertVC.switchDidChangeStateCallback = { on in
      print("on = \(on)")
    }
    
    confirmationAlertVC.noCallback = {
      print("no button did tap")
    }
    
    confirmationAlertVC.yesCallback = {
      print("yes button did tap")
    }
    self.presentViewController(confirmationAlertVC, animated: true, completion: nil)
  }

}

